source/applications/advanced/color_balance.py

```diff
@@ -0,0 +1,214 @@
+"""
+Balance color for RGB image.
+"""
+
+import datetime
+import numpy as np
+import matplotlib.pyplot as plt
+import zivid
+
+
+def _display_rgb(rgb, title):
+    """Display RGB image.
+
+    Args:
+        rgb: RGB image
+        title: Image title
+
+    Returns None
+
+    """
+    plt.figure()
+    plt.imshow(rgb)
+    plt.title(title)
+    plt.show(block=False)
+
+
+def _set_settings(dimension, iris, exposure_time, brightness, gain):
+    """Set settings for capture (3D or 2D).
+
+    Args:
+        dimension: '3d' or '2d'
+        iris: Iris
+        exposure_time: Exposure time
+        brightness: Projector brightness
+        gain: Gain
+
+    Returns:
+        settings: Capture settings (3D or 2D)
+
+    Raises:
+        ValueError: If dimension is not '3d' or '2d'
+
+    """
+    if dimension == "3d":
+        settings = zivid.Settings()
+        settings.iris = iris
+        settings.exposure_time = datetime.timedelta(microseconds=exposure_time)
+        settings.brightness = brightness
+        settings.gain = gain
+    elif dimension == "2d":
+        settings = zivid.Settings2D()
+        settings.iris = iris
+        settings.exposure_time = datetime.timedelta(microseconds=exposure_time)
+        settings.brightness = brightness
+        settings.gain = gain
+    else:
+        raise ValueError("The dimension value should be '3d' or '2d'")
```

Comment:
You also probably want to write what you got in: `, got: {dimension}`

Reply:
Done